astropy/coordinates/sky_coordinate.py

```diff
@@ -1490,6 +1490,10 @@ def _parse_coordinate_arg(coords, frame, units, init_kwargs):
                                      "frames: {0} != {1}".format(sc, scs[0]))
                 if allunitsphrepr and not isinstance(sc.data, UnitSphericalRepresentation):
                     allunitsphrepr = False
+            # special case of the above logic for edge case where we have a list of one
+            # SkyCoord
+            if len(scs) == 1:
+                allunitsphrepr = isinstance(scs[0].data, UnitSphericalRepresentation)
```

Comment:
It would seem simpler to just move this to l.1486 (replacing `allunitsphrepr = True` with this). That would also make the stanza correct for the case of a list where the first coordinate is not on the unit sphere.